backend/src/packages/trucks/truck.repository.ts

```diff
@@ -0,0 +1,132 @@
+import { eq, placeholder, sql } from 'drizzle-orm';
+
+import { DatabaseError } from '~/libs/exceptions/exceptions.js';
+import { type IRepository } from '~/libs/interfaces/interfaces.js';
+import { type IDatabase } from '~/libs/packages/database/database.js';
+import { type DatabaseSchema } from '~/libs/packages/database/schema/schema.js';
+
+import { type TruckEntity, type TruckEntityT } from './libs/types/types.js';
+
+class TruckRepository implements IRepository {
+  private db: Pick<IDatabase, 'driver'>;
+
+  private trucksSchema: DatabaseSchema['trucks'];
+
+  public constructor(
+    database: Pick<IDatabase, 'driver'>,
+    trucksSchema: DatabaseSchema['trucks'],
+  ) {
+    this.db = database;
+    this.trucksSchema = trucksSchema;
+  }
+
+  public async findById(id: number): Promise<TruckEntity[]> {
+    try {
+      return await this.db
+        .driver()
+        .select()
+        .from(this.trucksSchema)
+        .where(eq(this.trucksSchema.id, id));
+    } catch (error: unknown) {
+      throw new DatabaseError({
+        message: `Truck with ID ${id} was not found in the database`,
+        cause: error,
+      });
+    }
+  }
+
+  public async findAll(): Promise<TruckEntity[]> {
+    try {
+      return await this.db.driver().select().from(this.trucksSchema);
+    } catch (error: unknown) {
+      throw new DatabaseError({
+        message: 'Failed to retrieve trucks',
+        cause: error,
+      });
+    }
+  }
+
+  public async create(
+    entity: Omit<TruckEntityT, 'id'>,
+  ): Promise<TruckEntity[]> {
+    try {
+      const preparedInsert = this.db
+        .driver()
+        .insert(this.trucksSchema)
+        .values(entity)
+        .returning()
+        .prepare('createTruck');
+
+      return await preparedInsert.execute();
+    } catch (error: unknown) {
+      throw new DatabaseError({
+        message: 'Failed to create a new truck',
+        cause: error,
+      });
+    }
+  }
+
+  public async update(
+    id: number,
+    payload: Partial<TruckEntity>,
+  ): Promise<TruckEntity[]> {
+    try {
+      const preparedUpdate = this.db
+        .driver()
+        .update(this.trucksSchema)
+        .set(payload)
+        .where(sql`${this.trucksSchema.id} = ${placeholder('id')}`)
```

Comment:
```suggestion
        .where(eq(this.trucksSchema.id, placeholder('id')))
```
won't it work like this?